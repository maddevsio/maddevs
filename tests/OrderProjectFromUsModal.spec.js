import { mount, createLocalVue } from '@vue/test-utils'
import OrderProjectFromUsModal from '@/components/Modals/order-project-from-us-modal'
import Vuelidate from 'vuelidate'

const localVue = createLocalVue()

localVue.use(Vuelidate)

describe('order Project From Us Modal', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(OrderProjectFromUsModal, {
      localVue,
      stubs: ['modal'],
      mocks: {
        $store: {
          dispatch: () => new Promise(resolve => resolve({ status: 200 })),
        },
        $nuxt: {
          $emit: jest.fn(),
          $on: jest.fn(),
        },
      },
    })
    wrapper.vm.$refs = {
      checkboxes: {
        reset: jest.fn(),
      },
      form: {
        reset: jest.fn(),
      },
    }
  })

  // ------ IMPORTANT ----- //
  it('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  // --------------------- //

  it('sets the correct default data', () => {
    expect(typeof OrderProjectFromUsModal.data).toBe('function')
    const defaultData = OrderProjectFromUsModal.data()
    expect(defaultData.agreeWithPrivacyPolicy && defaultData.agreeToGetMadDevsDiscountOffers).toEqual(false)
    expect(defaultData.inputId).toEqual('order-project-from-us')
  })

  it('has a functions', () => {
    expect(
      typeof OrderProjectFromUsModal.methods.getPrivacyCheckboxState &&
        typeof OrderProjectFromUsModal.methods.getDiscountOffersCheckboxState,
    ).toBe('function')
  })

  it('call functions with params and change variables state', () => {
    wrapper.vm.getPrivacyCheckboxState(true)
    wrapper.vm.getDiscountOffersCheckboxState(true)

    expect(wrapper.vm.$data.agreeWithPrivacyPolicy && wrapper.vm.$data.agreeToGetMadDevsDiscountOffers).toEqual(true)
  })

  it('sendForm should add new object in $data.form', () => {
    const form = {
      templateId: 304632,
      variables: {
        agreeToGetMadDevsDiscountOffers: 'No',
        agreeWithPrivacyPolicy: 'No',
        email: '',
        emailTo: '',
        fullName: '',
        subject: 'Marketing',
        phoneNumber: '',
        company: '',
        projectDescription: '',
        modalTitle: 'Mad Devs Website Forms',
      },
    }
    expect(wrapper.vm.$data.form).toEqual('')
    wrapper.vm.sendForm(true)
    expect(wrapper.vm.$data.form).toEqual(form)
  })

  it('should rest values in data instances', () => {
    // Set mock data for data instances
    wrapper.vm.$data.fullName = 'Name'
    wrapper.vm.$data.email = 'email@mail.com'
    wrapper.vm.$data.projectDescriber = 'Project Describer'
    wrapper.vm.$data.form = {
      value1: 'value1',
      value2: 'value2',
    }
    wrapper.vm.$data.projectDescription = 'test'
    wrapper.vm.$data.company = 'test'
    wrapper.vm.$data.agreeWithPrivacyPolicy = true
    wrapper.vm.$data.agreeToGetMadDevsDiscountOffers = true
    wrapper.vm.$data.isEmailSent = true

    wrapper.vm.resetForm()
    expect(
      wrapper.vm.$data.fullName &&
        wrapper.vm.$data.email &&
        wrapper.vm.$data.form &&
        wrapper.vm.$data.projectDescription &&
        wrapper.vm.$data.company,
    ).toBeNull()
    expect(
      wrapper.vm.$data.agreeWithPrivacyPolicy &&
        wrapper.vm.$data.agreeToGetMadDevsDiscountOffers &&
        wrapper.vm.$data.isEmailSent,
    ).toEqual(false)
  })
})
