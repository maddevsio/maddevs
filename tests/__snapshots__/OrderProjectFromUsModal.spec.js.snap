// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order Project From Us Modal renders correctly 1`] = `
<div
  class="form"
>
  <div
    class="fields-list"
  >
    <div
      class="modal-field-item field-item"
    >
      <p
        class="modal-field-name field-name required"
      >
        Full Name
      </p>
       
      <input
        class="modal-entry-field entry-field"
        placeholder="John Smith"
        type="text"
      />
       
      <!---->
    </div>
     
    <div
      class="modal-field-item field-item"
    >
      <p
        class="modal-field-name field-name required"
      >
        Company
      </p>
       
      <input
        class="modal-entry-field entry-field"
        placeholder="MyAwesomeCompany, Inc."
        type="text"
      />
       
      <!---->
    </div>
     
    <div
      class="modal-field-item field-item"
    >
      <p
        class="modal-field-name field-name required"
      >
        Work email
      </p>
       
      <input
        class="modal-entry-field entry-field"
        placeholder="your@mail.com"
        type="text"
      />
       
      <!---->
    </div>
     
    <div
      class="modal-field-item field-item"
    >
      <p
        class="modal-field-name field-name"
      >
        Phone number
      </p>
       
      <input
        class="modal-entry-field entry-field"
        placeholder="+1 23X XXX-XXXX"
        type="text"
      />
       
      <!---->
    </div>
     
    <div
      class="modal-field-item field-item"
    >
      <p
        class="modal-field-name field-name"
      >
        Project description
      </p>
       
      <textarea
        class="modal-entry-field entry-field textarea"
        placeholder="Describe your project..."
        type="text"
      />
       
      <!---->
    </div>
  </div>
   
  <div
    class="form-checkboxes"
  >
    <label
      class="form-checkbox-label required"
      for="privacy-policy-order-project-from-us"
    >
      I confirm that I have read and accepted
    
      <div>
        Mad Devs’
      </div>
       
      <a
        href="/privacy/"
        target="_blank"
      >
        Privacy Policy
      </a>
       
      <input
        class="form-checkbox-input"
        id="privacy-policy-order-project-from-us"
        name="Privacy policy"
        type="checkbox"
      />
       
      <div
        class="checkmark"
      />
    </label>
     
    <label
      class="form-checkbox-label"
      for="marketing-communications-order-project-from-us"
    >
      I agree to get
    
      <div>
        Mad Devs’
      </div>
      
    discount offers and other marketing communications.
    
      <input
        class="form-checkbox-input"
        id="marketing-communications-order-project-from-us"
        name="Marketing communications"
        type="checkbox"
      />
       
      <div
        class="checkmark"
      />
    </label>
  </div>
   
  <button
    class="ui-button modal-button ui-button--disabled"
  >
    
    Submit your project
  
  </button>
</div>
`;
